project.clj

```diff
@@ -39,6 +39,7 @@
                                org.apache.httpcomponents/httpclient
                                net.sourceforge.nekohtml/nekohtml
                                ring/ring-core]]
+                 [com.cemerick/url "0.1.1"]                           ; URL library
```

Comment:
don't need to add this. Just require `[ring.util.codec :as codec]` and do `codec/form-encode`